text/0000-slatepack.md

```diff
@@ -0,0 +1,391 @@
+  - A Message Authentication Code is generated on the `header` data which is all of the header values concatenated
+    - `encode(HMAC[HKDF["", "header"](message key)](header))`
+      - `encode(data)` is canonical base64 from RFC 4648 without padding.
+      - `HMAC[key](message)` is HMAC from RFC 2104 with SHA-256.
+      - `HKDF[salt, label](key)` is 32 bytes of HKDF from RFC 5869 with SHA-256.
+
+#### Payload (age Enryption)
+
+A binary serialized slate is encrypted according to the age encryption specification. The steps taken here follow age as closely as possible to avoid losing any security properties. An fairly well-review [age library in rust](https://github.com/str4d/rage/tree/master/age) is available to use for implementation.
+
+Any deviations in SlatePack encryption from the exact cryptography steps and decisions made in age are unintentional and should be corrected unless they are explicitly stated as a deviation from the cryptography decisions made by age.
+
+The expected deviations from the age specification are in the formatting and structure of the contents of the file output data (as opposed to the encryption technique itself or the actual binary encrypted payload).
+
+- `payload`
+  - `nonce || STREAM[HKDF[nonce, "payload"](message key)](plaintext)`
+    - `nonce` is `random(16)`
+    - `message key` is 128-bit master message key for each recipient
+    - `STREAM` is `ChaCha20-Poly1305` in 64KiB chunks and a nonce structure of 11 bytes of big endian counter, and 1 byte of last block flag (0x00 / 0x01)
+    - `HKDF[salt, label](key)` is 32 bytes of HKDF from RFC 5869 with SHA-256
+
+### Armor
+
+The payload that will be armored is a binary serialized `SlatePack` JSON object. Armor is `Framing` wrapped around a `SimpleBase58Check` encoded `Payload`.
+
+#### Framing
+
+Armor uses specific `Headers`, `Footers` and `Periods` as `Framing` to contain its `Payload`.
+
+- `Header`
+  - Supported Headers:
+    - `BEGINSLATEPACK`
+    - `ENDSLATEPACK X/Y`
+      - `Y` is the total number of message parts for a given multipart message and `X` is the number of the current partial message
+      - Only used if there is more than one message part
+  - Regex: `^[>\n\r\t ]*BEGINSLATEPACK[>\n\r\t ]*([1-9]\/[1-9])*$`
+
+
+- `Footer`
+  - Supported Footers
+    - `ENDSLATEPACK`
+    - `ENDSLATEPACK X/Y`
+      - `Y` is the total number of message parts for a given multipart message and `X` is the number of the current partial message
+      - Only used if there is more than one message part
+  - Regex: `^[>\n\r\t ]*ENDSLATEPACK[>\n\r\t ]*([1-9]\/[1-9])*$`
+
+- `Periods`
+
+  - All data of an armored slate up to the first `.` is the framing header
+  - All data after the first `.` and before the second `.` is the `SimpleBase58Check` encoded payload which contains the slate data
+  - All data after the second `.` and before the third `.` is the framing footer
+  - Any data after the third `.` is ignored
+
+
+#### `SimpleBase58Check` Encoding
```

Comment:
If we are supporting encryption-only from the start, there's no need for any error checking at all, since ChaCha20-Poly1305 already covers message authentication.

Reply:
There are some potential metadata fields that aren't encrypted that we still want some amount of integrity for: `version`, `mode`, `sender` etc.